ext/winevt/winevt_subscribe.c

```diff
@@ -144,17 +153,61 @@ rb_winevt_subscribe_subscribe(int argc, VALUE* argv, VALUE self)
   return Qfalse;
 }
 
+BOOL
+is_rate_limit_exceeded(struct WinevtSubscribe *winevtSubscribe)
+{
+  time_t now;
+  ULONG currRate = 0;
+
+  if (winevtSubscribe->rateLimit == SUBSCRIBE_RATE_INFINITE)
+    return FALSE;
+
+  time(&now);
+
+  winevtSubscribe->previousRate = 0;
+
+  if (now <= winevtSubscribe->lastTime) {
+    winevtSubscribe->previousRate = winevtSubscribe->currentRate;
+    currRate = winevtSubscribe->currentRate;
+    if (currRate >= winevtSubscribe->rateLimit) {
+      return TRUE;
+    }
+  }
+
+  return FALSE;
+}
+
+void
+rate_limit_state(struct WinevtSubscribe *winevtSubscribe, ULONG count)
```

Comment:
`rate_limit_state()` is still hard to understand what it's doing.
I prefer `update_rate_limit_state()`.

Reply:
Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/9/commits/585f5ef6ccf92e852c33b2ce5c25b7b1e15d09f9.